src/persistance/metadataWriter.ts

```diff
@@ -0,0 +1,20 @@
+import { Metadata } from "../metadata/MetadataGenerable";
+import { META_DATA_LOCATION } from "../constants";
+
+export default class MetadataWriter {
+  private _fs: any;
+
+  constructor(fs: any) {
+    this._fs = fs;
+  }
+
+  writeMetadata(metaData: Metadata) {
+    const fileName: string = metaData.type + ".dat";
+    const fileLocation = META_DATA_LOCATION + "/" + fileName;
```

Comment:
The fileLocation would be `META_DATA_LOCATION + "/" + metadata.type + "/" + metadata.id + ".dat"`